spec/ruby/core/binding/eval_spec.rb

```diff
@@ -13,6 +13,16 @@
     bind.eval("Inside.name").should == "BindingSpecs::Demo::Inside"
   end
 
+  it "does not leak variables to cloned bindings" do
+    obj = BindingSpecs::Demo.new(1)
+    bind = obj.get_empty_binding
+    bind2 = bind.dup
+
+    bind.eval("x = 72")
```

Comment:
I think it's important that both the eval path and local_variable_set path are exercised as they probably don't go through the same code path. Hence I put the eval case here and added local_variable_set stuff to the clone spec.